fast64_internal/oot/exporter/main.py

```diff
@@ -0,0 +1,143 @@
+import bpy
+import os
+
+from dataclasses import dataclass
+from mathutils import Matrix
+from bpy.types import Object
+from typing import Optional
+from ...f3d.f3d_gbi import DLFormat, TextureExportSettings
+from ..scene.properties import OOTBootupSceneOptions
+from ..scene.exporter.to_c import setBootupScene
+from ..oot_model_classes import OOTModel
+from ..oot_f3d_writer import writeTextureArraysNew
+from ..oot_level_writer import writeTextureArraysExistingScene
+from .scene import Scene
+from .other import Files
+from .classes import SceneFile
+
+from ...utility import (
+    PluginError,
+    checkObjectReference,
+    unhideAllAndGetHiddenState,
+    restoreHiddenState,
+    toAlnum,
+)
+
+from ..oot_utility import (
+    ExportInfo,
+    OOTObjectCategorizer,
+    ootDuplicateHierarchy,
+    ootCleanupScene,
+    getSceneDirFromLevelName,
+    ootGetPath,
+)
+
+
+@dataclass
+class SceneExporter:
```

Comment:
if this class had less members I'd consider this but I really don't think turning this into a function would be a good thing in this case, it would make a function with 11 parameters (which is too much imo, especially since OoT export is a complex thing), also some of this class members are used somewhere else (to get the scene name, I think it's better this way)

Reply:
I see, so the class is used both to export and to pass around export parameters. I think it would be better to rename it something like `SceneExport` or `SceneExportParams`, as that's how its used when passing it to Spec or Files. The naming is a little weird because the Spec/Files functions are technically part of the export process, but passing an class named Exporter to it implies that exporting is a sub aspect of those functions. I'm also noticing that the remove scene operator still uses the the old spec functions, not sure if that's intended.

Reply:
changed it to ``SceneExport``